lib/services/sf.js

```diff
@@ -782,6 +782,13 @@ function StateAbstract(options)
     requestOptions.headers =
       Util.apply(this.getDefaultReqHeaders(), requestOptions.headers || {});
 
+    if (Util.isLoginRequest(requestOptions.url)) {
```

Comment:
should this be applied only for login requests?

Reply:
Sorry, perhaps I asked the question in the wrong way. Should we apply additional headers only when the request is a login request? What about e.g. query requests?

Reply:
Sorry, I misunderstood your question. This strategy is only for the login requests according to the document. Please have a look at https://github.com/snowflakedb/snowflake-sdks-drivers-issues-teamwork/issues/575#issuecomment-1751273272.